src/overlays/actors/ovl_En_Changer/z_en_changer.c

```diff
@@ -20,15 +35,257 @@ const ActorInit En_Changer_InitVars = {
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Changer/EnChanger_Init.s")
+static Vec3f sLeftChestPos[] = {
+    { 0.0f, 0.0f, 0.0f },         { -100.0f, 20.0f, -245.0f },  { -100.0f, 20.0f, -685.0f },
+    { -100.0f, 20.0f, -1125.0f }, { -100.0f, 20.0f, -1565.0f }, { -100.0f, 20.0f, -2005.0f },
+};
+
+static Vec3f sRightChestPos[] = {
+    { 0.0f, 0.0f, 0.0f },        { 140.0f, 20.0f, -245.0f },  { 140.0f, 20.0f, -685.0f },
+    { 140.0f, 20.0f, -1125.0f }, { 140.0f, 20.0f, -1565.0f }, { 140.0f, 20.0f, -2005.0f },
+};
+
+static s32 sLoserGetItemIds[] = {
+    GI_NONE, GI_RUPEE_GREEN_LOSE, GI_RUPEE_GREEN_LOSE, GI_RUPEE_BLUE_LOSE, GI_RUPEE_BLUE_LOSE, GI_RUPEE_RED_LOSE,
+};
+
+static s32 sEtcParams[] = { 0x0000, 0x0008, 0x0008, 0x0009, 0x0009, 0x000A };
+
+static s32 sTreasureFlags[] = { 0x0000, 0x0002, 0x0004, 0x0006, 0x0008, 0x000A };
+
+void EnChanger_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+}
+
+void EnChanger_Init(Actor* thisx, GlobalContext* globalCtx2) {
+    EnChanger* this = THIS;
+    GlobalContext* globalCtx = globalCtx2;
+    s16 leftChestParams;
+    s16 rightChestParams;
+    s16 hpChestParams;
+    s16 minigameRoomNum;
+    s16 rightItemEtceteraOffset;
+    s16 leftItemParams;
+    s16 temp_v1_3;
+    s16 new_var;
+    s32 hpParams;
+
+    if (1) {}
+
+    minigameRoomNum = globalCtx->roomCtx.curRoom.num - 1;
+    if (minigameRoomNum < 0) {
+        minigameRoomNum = 0;
+    }
+    if (Flags_GetTreasure(globalCtx, sTreasureFlags[minigameRoomNum])) {
+        this->roomChestsOpened = true;
+    }
+
+    osSyncPrintf("\n\n");
+    // Treasure generation (which room is it?)
+    osSyncPrintf(VT_FGCOL(GREEN) "☆☆☆☆☆ 宝発生(部屋はどれ？) %d\n" VT_RST, globalCtx->roomCtx.curRoom.num);
+    // How is the Bit?
+    osSyncPrintf(VT_FGCOL(GREEN) "☆☆☆☆☆ ビットは？ \t     %x\n" VT_RST, globalCtx->actorCtx.flags.chest);
+    // How is the Save BIT?
+    osSyncPrintf(VT_FGCOL(GREEN) "☆☆☆☆☆ セーブＢＩＴは？     %x\n" VT_RST, sTreasureFlags[minigameRoomNum]);
+    // Is it already a zombie?
+    osSyncPrintf(VT_FGCOL(GREEN) "☆☆☆☆☆ もう、ゾンビ？\t     %d\n" VT_RST, this->roomChestsOpened);
+    osSyncPrintf("\n\n");
+
+    minigameRoomNum *= 2;
+    // Spawn Heart Piece in chest (or Purple Rupee if won Heart Piece)
+    if (globalCtx->roomCtx.curRoom.num >= 6) {
+        hpChestParams = ((gSaveContext.itemGetInf[1] & 0x800) ? (0x4EA0) : (0x4EC0));
+        hpChestParams = sTreasureFlags[5] | hpChestParams;
+        this->finalChest = (EnBox*)Actor_SpawnAsChild(&globalCtx->actorCtx, &this->actor, globalCtx, ACTOR_EN_BOX,
+                                                      20.0f, 20.0f, -2500.0f, 0, 0x7FFF, 0, hpChestParams);
+        if (this->finalChest != NULL) {
+            if (this->roomChestsOpened) {
+                Flags_SetTreasure(globalCtx, hpChestParams & 0x1F);
+                Actor_Kill(&this->actor);
+                return;
+            } else {
+                hpParams = ((gSaveContext.itemGetInf[1] & 0x800) ? (0xB) : (0xC)) & 0xFF;
```

Comment:
The 0xB and 0xC here should probably also be values from the `ItemEtceteraType` enum.